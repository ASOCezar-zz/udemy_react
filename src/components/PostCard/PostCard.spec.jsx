import { render, screen } from '@testing-library/react';
import { PostCard } from '.';

const mockPostCard = {
  title: 'Title1',
  body: 'Body1',
  id: 1,
  cover: 'img/img.png',
};

describe('<PostCard />', () => {
  it('should render PostCard correctly', () => {
    render(<PostCard {...mockPostCard} />);

    expect(screen.getByRole('img', { name: mockPostCard.title })).toHaveAttribute('src', mockPostCard.cover);

    expect(screen.getByRole('heading', { name: 'Title1 1' })).toBeInTheDocument();
    expect(screen.getByText(mockPostCard.body)).toBeInTheDocument();
  });

  it('should match with snapshot', () => {
    const { container } = render(<PostCard {...mockPostCard} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
