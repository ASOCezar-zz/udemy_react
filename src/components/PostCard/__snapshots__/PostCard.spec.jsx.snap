// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match with snapshot 1`] = `
<div
  class="sc-bdnxRM bkXcNm post"
>
  <img
    alt="Title1"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      Title1
       
      1
    </h2>
    <hr />
    <span>
      Body1
    </span>
  </div>
</div>
`;
